CppCoreGuidelines.md

```diff
@@ -14316,14 +14316,30 @@ This has no effect on synchronization: The data race is still there!
 The C++ mechanism for this is `atomic` types:
 
     atomic<int> free_slots = max_slots; // current source of memory for objects
-
+    void check(int n)
+    {
+        // will throw on error 
+        // ...
+    }
     Pool* use()
     {
-        if (int n = free_slots--) return &pool[n];
+        // Apply the Copy-Validate-Use strategy:
+        // 1. Atomic substraction + local copy:
+        int n = free_slots.fetch_sub(1); // Use atomic operations explicitly
+        // 2. Validate n: 
+        check(n);
```

Comment:
The ```check()``` is a huge improvement.  The current example doesn't return anything for the ```else``` case and the caller has no way to know.

Maybe more contex could be added to the example too?  Is ```use()``` in a class?  It seems odd to have a function named 'use' that takes no parameters and returns a pointer.

Reply:
Thank you for the positive feedback. 

> The `check()` is a huge improvement. The current example doesn't return anything for the `else` case and the caller has no way to know.
> 
> Maybe more contex could be added to the example too? Is `use()` in a class? It seems odd to have a function named 'use' that takes no parameters and returns a pointer.

Thank you for the positive feedback. I just wanted to add some community feedback so the editors (or other cpp-masters) can review  the 'door-opener' examples of the CP-section. A returning pointer really distracts, maybe just a 'simple' cout << would do , so one does not have to reason about 'dangling' pointers. The 'Every-Stack-Has-Its-Own-Stack' feature is much more worthy to think about especially in terms of thread-safety of (recursive) functions and the implications on resource consumption of threads.